src/storm/mesos/logviewer/LogViewerController.java

```diff
@@ -0,0 +1,180 @@
+    }
+    
+    protected Process process;
+    protected IUrlDetection urlDetector;
+    protected String installLocation;
+    protected Map<String,Object> config;
+    protected Optional<Integer> port;
+    protected static final String LOG_VIEWER_PORT = "logviewer.port";
+       
+    protected void setProcess(Process process) {
+        this.process = process;
+    }
+    
+    protected Process getProcess() {
+        return process;
+    }
+    
+    protected void setPort(Optional<Integer> oport) {
+        port = oport;
+    }
+    
+    protected Optional<Integer> getPort() {
+        return port;
+    }
+    
+    protected void setConfig(Map<String,Object> configuration) {
+        config = configuration;
+    }
+    
+    protected Map<String,Object> getConfig(){
+        return config;
+    }
+    
+    protected void launchLogViewer () throws IOException {
+         
+        if(LOG.isDebugEnabled())
+            LOG.debug("launchLogViewer: home dir? " + getInstallLocation());
+        
+        ProcessBuilder pb = createProcessBuilder(getInstallLocation());
+        setProcess(pb.start());       
+    }
+    
+    /**
+     * Create a process builder to launch the log viewer
+     * @param logDirectory
+     * @return
+     */
+    protected ProcessBuilder createProcessBuilder(String homeDirectory) {
+        ProcessBuilder pb = new ProcessBuilder(Paths.get(homeDirectory,"/bin/storm").toString(),"logviewer");
+         
+        // If anything goes wrong at startup we want to see it.
+        File log = Paths.get(homeDirectory,"/logs/logviewer-startup.log").toFile();
+        pb.redirectErrorStream(true);
+        pb.redirectOutput(Redirect.appendTo(log)); 
+        return pb;
+    }
+    
+    @SuppressWarnings("unchecked")
+    protected void initialize() {
+        setInstallLocation(new File(".").getAbsolutePath());
+        
+        loadConfig();
+        
+        // If the storm configuration has defined a port use it.
+        if (getPort().isPresent()) {
+            getUrlDetector().setPort(getPort());
+        }
+        else {
+            // Otherwise use the default 
+            LOG.info("No valid port in storm.yaml, using the on configured in the Url Detector " + 
+                getUrlDetector().getPort());
+            setPort(getUrlDetector().getPort());
+        }        
+    }
+    
+    @SuppressWarnings("unchecked")
+    protected void loadConfig() {
+        InputStream input = null;
+        try {
+            input = new FileInputStream(Paths.get(getInstallLocation(),"/conf/storm.yaml").toFile());
+            setConfig((Map<String, Object>)new Yaml().load(input));
+            LOG.info("Setting port for log viewer to " + config.get(LOG_VIEWER_PORT));
+            setPort(Optional.of((Integer)getConfig().get(LOG_VIEWER_PORT)));
+        } catch (Exception e) {
+            LOG.error("Failed to configure the log viewer controller,", e);
+            setPort(Optional.fromNullable(((Integer)null)));
```

Comment:
This was actually simply handling the exception thrown by FileInputStream. Setting the port is more to insure the Optional was set correctly, not so much the "port" was set. The Optional thing was because folks here love Optionals.....I'm not a big fan and will happily remove them.